stac_ipyleaflet/widgets/inspect.py

```diff
@@ -0,0 +1,129 @@
+                if "/cog" in layer.url:
+                    parsed_url = urlparse(layer.url)
+                    parsed_query = parse_qs(parsed_url.query)
+                    url = parsed_query["url"][0]
+                    data = make_get_request(f"{cog_partial_request_path}{url}")
+                    if data:
+                        visible_layers_data.append(
+                            {"layer_name": layer.name, "data": data.json()}
+                        )
+                # @TODO: Add logic to grab point data for "/mosaics" layers here
+                # @NOTE: Blocked until Impact Titiler is updated to access /mosaicsjson/point
+            return visible_layers_data
+
+        def display_data(layers_data: LayerData):
+            point_data.value = ""
+
+            def create_html_template(layer_name, coordinates, values, band_names):
+                template = f"""
+                    <p>
+                        <b>
+                            {layer_name}
+                        </b>
+                    </p>
+                    <ul>
+                        <li>
+                            Coordinates: {coordinates}
+                        </li>
+                        <li>
+                            Values: {values}
+                        </li>
+                        <li>
+                            Bands: {band_names}
+                        </li>
+                    </ul>
+                """
+                return template
+
+            for layer in layers_data:
+                point_data.value += create_html_template(
+                    layer["layer_name"],
+                    layer["data"]["coordinates"],
+                    layer["data"]["values"],
+                    layer["data"]["band_names"],
+                )
+            return
+
+        def handle_interaction(self, action, geo_json, **kwargs):
+            def handle_clear(event):
+                draw_layer = main.find_layer("draw_layer")
+                main.remove_layer(draw_layer)
+                point_data.value = "<code>Waiting for points of interest...</code>"
+                clear_button.disabled = True
+                return
+
+            self.coordinates = []
+
+            if action == "created":
+                if geo_json["geometry"] and geo_json["geometry"]["type"] == "Point":
+                    geojson_layer = GeoJSON(
+                        name="draw_layer",
+                        data=geo_json,
+                    )
+                    main.add_layer(geojson_layer)
+                    self.coordinates = geo_json["geometry"]["coordinates"]
+
+                    if len(applied_layers):
+                        layers_data = get_visible_layers_data(self.coordinates)
+                        if layers_data:
+                            display_data(layers_data)
+                        else:
+                            point_data.value = f"<p><b>Coordinates:</b></p><code>{self.coordinates}</code><br/>"
```

Comment:
Any thoughts on handling in the same html_template pattern as the rest of the data? Not a huge deal, but just an idea to keep things in a similar style.